game.go

```diff
@@ -770,18 +771,33 @@ func SchedNow() int {
 	return 0
 }
 
+func runMain(sprite Sprite) {
+	isSchedInMain = true
+	lastSched = time.Now()
+	sprite.Main()
+	isSchedInMain = false
+}
+
 func Sched() int {
 	now := time.Now()
-	if now.Sub(lastSched) >= 3e7 {
-		if me := gco.Current(); me != nil {
-			gco.Sched(me)
+	if isSchedInMain {
+		if now.Sub(lastSched) >= time.Second*3 {
+			lastSched = now
+			panic("Loop execution timed out. Please check if there is an infinite loop in the code.")
```

Comment:
```suggestion
			panic("Main execution timed out. Please check if there is an infinite loop in the code.")
```

I think we are not sure if there's really a loop in code.